File: php/classes/constructors_and_destructors/constructor_with_initialisation.php
#!/usr/bin/php8.1
<?php

class Basket {
  public function __construct() {
    $this->capacity = 3;
    $this->load_counter = 0;
  }

  public function put(string $item): void {
    if ($this->load_counter < $this->capacity) {
      $this->items[$this->load_counter] = $item;
    }
    $this->load_counter++;
  }

  public function pull(): string {
    $this->item = $this->items[$this->load_counter];
    $this->items[$this->load_counter] = "";
    $this->load_counter--;

    return $this->item;
  }

  public function show(): void {
    foreach ($this->items as $item) {
      print($item . " ");
    }
    print("\n");
  }

  private int $capacity;
  private array $items = [];
  private int $load_counter;
}

$handbasket = new Basket();

$handbasket->put("apple");
$handbasket->put("pear");
$handbasket->put("plum");

$handbasket->show();

File: python/classes/constructors_and_destructors/constructor_with_initialisation.py
#!/usr/bin/python3

class Basket:
    def __init__(self):
        self.capacity = 3
        self.items = []
        self.load_counter = 0

    def put(self, item):
        if (self.load_counter < self.capacity):
            self.items.append(item)

        self.load_counter += 1

    def pull(self):
        item = self.items[self.load_counter]
        self.items[self.load_counter] = ""
        self.load_counter -= 1

    def show(self):
        for item in self.items:
            print(item + " ", end = "")

        print()

handbasket = Basket()

handbasket.put("apple")
handbasket.put("pear")
handbasket.put("plum")

handbasket.show()
